### PR TITLE
Add geolocation button

### PR DESCRIPTION
**Pull request by hwbllmnn**
### Description:

This PR adds a geolocation button.

@terrestris/devs please review

### CODE REVIEW

File: src/Button/GeoLocationButton/GeoLocationButton.jsx [hunk 1]
```diff
@@ -0,0 +1,265 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import OlMap from 'ol/map';
+import OlGeolocation from 'ol/geolocation';
+import OlGeomLineString from 'ol/geom/linestring';
+import OlOverlay from 'ol/overlay';
+
+import ToggleButton from '../ToggleButton/ToggleButton.jsx';
```
Comment (dnlkoch) [inline src/Button/GeoLocationButton/GeoLocationButton.jsx]: You may want to import the `ToggleButton` from the `index.js`.

File: src/Button/GeoLocationButton/GeoLocationButton.example.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+render(
+  <div>
+    <div id="map" style={{
+      height: '400px'
+    }} />
+
+    <div className="example-block">
+      <ToggleGroup>
+        <GeoLocationButton
+          onGeolocationChange={() => undefined}
```
Comment (KaiVolland) [inline src/Button/GeoLocationButton/GeoLocationButton.example.jsx]: Can be removed.

File: src/Button/GeoLocationButton/GeoLocationButton.jsx [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+    const view = map.getView();
+
+    // Geolocation Control
+    this.geolocationInteraction = new OlGeolocation({
+      projection: view.getProjection(),
+      trackingOptions: {
+        maximumAge: 10000,
+        enableHighAccuracy: true,
+        timeout: 600000
+      }
```
Comment (dnlkoch) [inline src/Button/GeoLocationButton/GeoLocationButton.jsx]: `trackingOptions` should be configurable via a `prop`.

File: src/Button/GeoLocationButton/GeoLocationButton.jsx [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+import OlGeolocation from 'ol/geolocation';
+import OlGeomLineString from 'ol/geom/linestring';
+import OlOverlay from 'ol/overlay';
+
+import ToggleButton from '../ToggleButton/ToggleButton.jsx';
+import { MathUtil } from '../../index';
+
+import './GeoLocationButton.less';
+import mapMarker from '../../../assets/geolocation-marker.png';
+import mapMarkerHeading from '../../../assets/geolocation-marker-heading.png';
```
Comment (KaiVolland) [inline src/Button/GeoLocationButton/GeoLocationButton.jsx]: I'd prefer to store the assets of a component within its component folder to increase the cohesion. Any thoughts?

File: src/Button/GeoLocationButton/GeoLocationButton.jsx [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+      if (heading && speed) {
+        this.markerEl.src = mapMarkerHeading;
+      } else {
+        this.markerEl.src = mapMarker;
+      }
+    }
+
+    this.updateView();
+
+    this.props.onGeolocationChange({
```
Comment (KaiVolland) [inline src/Button/GeoLocationButton/GeoLocationButton.jsx]: Ich like passing an object here! :top:

File: src/Button/GeoLocationButton/GeoLocationButton.less [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  border-left: 6px solid transparent;
+  content: "";
+  position: absolute;
+  bottom: -6px;
+  margin-left: -7px;
+  left: 50%;
+}
+.react-geo-measure-tooltip-static:before {
+  border-top-color: @measure-static-tooltip-background-color;
+}
```
Comment (KaiVolland) [inline src/Button/GeoLocationButton/GeoLocationButton.less]: Ich think the measure stuff can be removed?!

File: src/Button/GeoLocationButton/GeoLocationButton.spec.jsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+      expect(GeoLocationButton).not.toBeUndefined();
+    });
+
+    it('can be rendered', () => {
+      const wrapper = TestUtil.mountComponent(GeoLocationButton, {
+        map: map
+      });
+      expect(wrapper).not.toBeUndefined();
+    });
+
```
Comment (KaiVolland) [inline src/Button/GeoLocationButton/GeoLocationButton.spec.jsx]: Plz add more tests.

File: src/Button/GeoLocationButton/GeoLocationButton.jsx (hunk 1 above)
Comment (hwbllmnn) [reply src/Button/GeoLocationButton/GeoLocationButton.jsx]: Done.

File: src/Button/GeoLocationButton/GeoLocationButton.jsx (hunk 3 above)
Comment (hwbllmnn) [reply src/Button/GeoLocationButton/GeoLocationButton.jsx]: Done

File: src/Button/GeoLocationButton/GeoLocationButton.less (hunk 6 above)
Comment (hwbllmnn) [reply src/Button/GeoLocationButton/GeoLocationButton.less]: True, nice catch!

File: src/Button/GeoLocationButton/GeoLocationButton.spec.jsx (hunk 7 above)
Comment (hwbllmnn) [reply src/Button/GeoLocationButton/GeoLocationButton.spec.jsx]: Ok, I've added a few more tests.

**Comment by KaiVolland** [comment]
Good job. Plz merge.

**Pull request merged**